tests/wpunit/Admin/Fields/FormTest.php

```diff
@@ -0,0 +1,198 @@
+<?php
+
+namespace StellarWP\Uplink\Admin\Fields;
+
+use StellarWP\Uplink\Config;
+use StellarWP\Uplink\Register;
+use StellarWP\Uplink\Resources\Collection;
+use StellarWP\Uplink\Tests\UplinkTestCase;
+use StellarWP\Uplink\Tests\TestUtils;
+use tad\Codeception\SnapshotAssertions\SnapshotAssertions;
+use StellarWP\Uplink\Tests\Traits\With_Uopz;
+use StellarWP\Uplink\View\WordPress_View;
+use StellarWP\Uplink as UplinkNamespace;
+
+class FormTest extends UplinkTestCase {
+
+	use TestUtils;
+	use SnapshotAssertions;
+	use With_Uopz;
+
+	protected $view;
+
+	protected $collection;
+
+	public function setUp(): void {
+		parent::setUp();
+
+		$this->view = $this->container->get( WordPress_View::class );
+
+		$this->collection = $this->container->get( Collection::class );
+	}
+
+	public function setup_container_get_slug( $resource ) {
+		$collection = Config::get_container()->get( Collection::class );
+
+		Register::{$resource['type']}(
+			$resource['slug'],
+			$resource['name'],
+			$resource['version'],
+			$resource['path'],
+			$resource['class']
+		);
+
+		return $collection->get( $resource['slug'] );
+	}
+
+	public function resourceProvider() {
+		$resources = $this->get_test_resources();
+
+		foreach ( $resources as $resource ) {
+			yield $resource['slug'] => [ $resource ];
+		}
+	}
+
+	/**
+	 * @test
+	 * @dataProvider resourceProvider
+	 */
+	public function it_should_add_field_to_form( $resource ) {
+		$current_resource = $this->setup_container_get_slug( $resource );
+		$slug             = $current_resource->get_slug();
+
+		$field = UplinkNamespace\get_field( $current_resource->get_slug() );
+		$field->set_field_name( 'field-' . $slug );
+		$field->show_label( true );
+
+		$form = new Form( $this->view );
+		$form->add_field( $field );
+
+		$fields = $form->get_fields();
+		$this->assertArrayHasKey( $slug, $fields );
+		$this->assertSame( $field, $fields[ $slug ] );
+	}
+
+	/**
+	 * @test
+	 * @dataProvider resourceProvider
+	 */
+	public function it_should_set_and_get_button_text( $resource ) {
+		$button_text = 'Submit';
+		$form        = new Form( $this->view );
+		$form->set_button_text( $button_text );
+
+		$this->assertEquals( $button_text, $form->get_button_text() );
+	}
+
+	/**
+	 * @test
+	 * @dataProvider resourceProvider
+	 */
+	public function it_should_render_form( $resource ) {
+		$current_resource = $this->setup_container_get_slug( $resource );
```

Comment:
Again, I might not be correct, but the user is `0` (not logged-in); should the form be added/displayed in this case?